modules/cudacodec/src/video_parser.cpp

```diff
@@ -78,9 +78,18 @@ bool cv::cudacodec::detail::VideoParser::parseVideoData(const unsigned char* dat
     if (rawMode)
         currentFramePackets.push_back(RawPacket(data, size, containsKeyFrame));
 
-    if (cuvidParseVideoData(parser_, &packet) != CUDA_SUCCESS)
-    {
-        CV_LOG_ERROR(NULL, "Call to cuvidParseVideoData failed!");
+    CUresult retVal = CUDA_SUCCESS;
+    try {
```

Comment:
@asmorkalov I had to change this because it appears [cuvidParseVideoData() does not propagate error codes](https://forums.developer.nvidia.com/t/errors-not-propagated-to-cuvidparsevideodata/259856) so the only way to signal failure is with an exception.